core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -372,12 +453,28 @@ object SparkSubmit {
       sysProps("spark.submit.pyFiles") = formattedPyFiles
     }
 
+    // NOTE: If we are using the REST gateway, we will use the original arguments directly.
```

Comment:
I'm very confused by this comment. At this point, we don't actually know whether it will use REST or not, correct?